packages/react-search-ui-views/src/SearchBox.js

```diff
@@ -53,25 +56,35 @@ function SearchBox(props) {
                 appendClassName("sui-search-box", className) + autocompleteClass
               }
             >
-              <div className="sui-search-box__wrapper">
-                <input
-                  {...getInputProps({
+              <InputView
+                getInputProps={additionalProps =>
+                  getInputProps({
                     placeholder: "Search your documents",
                     ...inputProps,
                     className: `${appendClassName(
                       "sui-search-box__text-input",
                       inputProps.className
-                    )} ${focusedClass}`
-                  })}
-                />
-                {useAutocomplete && isOpen && allAutocompletedItemsCount > 0 ? (
-                  <AutocompleteView {...props} {...downshiftProps} />
-                ) : null}
-              </div>
-              <input
-                type="submit"
-                value="Search"
-                className="button sui-search-box__submit"
+                    )} ${focusedClass}`,
+                    ...additionalProps
+                  })
+                }
+                getButtonProps={additionalProps => ({
+                  type: "submit",
+                  value: "Search",
+                  className: "button sui-search-box__submit",
+                  ...additionalProps
+                })}
+                getAutocomplete={() => {
+                  if (
+                    useAutocomplete &&
+                    isOpen &&
+                    allAutocompletedItemsCount > 0
+                  ) {
+                    return <AutocompleteView {...props} {...downshiftProps} />;
+                  } else {
+                    return null;
```

Comment:
Very minor code style suggestion - but you might get fewer lines/a smoother flow by turning this into a ternary: 

```jsx
getAutocomplete={() => {
  const showAutocomplete =
    (useAutocomplete && isOpen && allAutocompletedItemsCount > 0);
  return showAutocomplete
    ? <AutocompleteView {...props} {...downshiftProps} />
    : null;
}}
```

Reply:
Sometimes the explicit if / else is just easier to read. I don't mind changing it though.

Reply:
I'm good either way!